tensilelite/Tensile/SolutionStructs.py

```diff
@@ -3960,20 +3960,26 @@ def calcEpilogueTurns(factorDims: List) -> int:
     # Requires preciseBounds check since we rely on the buffer bounds check, not
     # individual vector registers doing bounds compares.
 
-    if state["_UseSgprForGRO"] == -1:
+    if state["_UseSgprForGRO"] != 0:
       # Don't use SGPR if it looks like we might not have enough - better to leave PBC enabled even if we have to use VGPR
       # 40 is based on current SGPR usage, this may need to be tuned in the future:
       numLoadsA = state["NumLoadsCoalescedA"]*state["NumLoadsPerpendicularA"]
       numLoadsB = state["NumLoadsCoalescedB"]*state["NumLoadsPerpendicularB"]
       numLoadsM = 0
       if state["ProblemType"]["Sparse"] and not state["DirectToVgprSparseMetadata"]:
         numLoadsM = state["NumLoadsCoalescedMetadata"]*state["NumLoadsPerpendicularMetadata"]
-      if numLoadsA + numLoadsB + numLoadsM > 35 or state["DirectToVgprA"] or state["DirectToVgprB"]: # force _UseSgprForGRO = 0 if DirectToVgpr is enabled
-        #print "info: Disabling UseSgprForGRO since predicting too many SGPR will be used"
-        state["_UseSgprForGRO"] = 0
-      else:
-        state["_UseSgprForGRO"] = 1
+      if state["_UseSgprForGRO"] == -1:
+        if numLoadsA + numLoadsB + numLoadsM > 35 or state["ProblemType"]["SwizzleTensorA"] or state["ProblemType"]["SwizzleTensorB"]:
+          #print "info: Disabling UseSgprForGRO since predicting too many SGPR will be used"
+          state["_UseSgprForGRO"] = 0
+        else:
+          state["_UseSgprForGRO"] = 1
 
+      if state["_UseSgprForGRO"] == 1:
+        if numLoadsA + numLoadsB + numLoadsM > 35:
+          reject(state, "Too many SGPR will be used for UseSgprForGRO")
```

Comment:
We don't need to reject it in SolutionStruct. If it uses too many sgpr, it should be rejected when checking vgpr usage in final stage.

Reply:
fixed, thx!